Added Trufflehog

GPelayo opened the pull request:
The addition to the gitlab script is only a stand-in until testing architecture is complete.

Commit:
Added Trufflehog

hannes-ucsc commented on .gitlab-ci.lower-env.yml:
```diff
@@ -24,6 +24,17 @@ cache:
+  stage: build
+  image: python:3.5
+  before_script:
+    - echo pip install trufflehog
```
```suggestion
    - pip install trufflehog
```

hannes-ucsc commented on .gitlab-ci.lower-env.yml:
```diff
@@ -24,6 +24,17 @@ cache:
+  before_script:
+    - echo pip install trufflehog
+  script:
+    - echo trufflehog --regex --rules trufflehog-rules.json --branch develop --entropy=False --max_depth 30 https://github.com/HumanCellAtlas/data-portal.git
```
```suggestion
    - trufflehog --regex --rules trufflehog-rules.json --branch develop --entropy=False --max_depth 30 https://github.com/HumanCellAtlas/data-portal.git
```

hannes-ucsc reviewed (changes requested):
And just to confirm, you mentioned that Trufflehog actually found a secret in the code base? Why isn't that being detected in https://allspark.dev.data.humancellatlas.org/HumanCellAtlas/data-portal/-/jobs/29602 where the `echo` was absent?

GPelayo commented:
~~I probably meant the data-browser codebase. The data-portal didn't contain any secrets.~~ The secret I found was in the data-browser codebase and it popped up before I added the argument `--max_depth 30` to the Trufflehog command. I added this arg because Azul only checks secrets in the 30 most recent commits and I want to keep the Trufflehog configurations consistent.

hannes-ucsc commented:
@GPelayo, please Slack (!) me the commit and location of that secret.
